network/src/commonMain/kotlin/com/wire/kalium/network/api/notification/NotificationResponse.kt

```diff
@@ -1,35 +1,11 @@
 package com.wire.kalium.network.api.notification
 
 import kotlinx.serialization.SerialName
-import kotlinx.serialization.Serializable
+import kotlinx.serialization.Transient
 
-sealed class NotificationResponse {
-    abstract val time: String
-    abstract val hasMore: Boolean
-    abstract val notifications: List<EventResponse>
-
-    data class MissingSome(override val time: String, override val hasMore: Boolean, override val notifications: List<EventResponse>) :
-        NotificationResponse() {
-        internal constructor(notificationPage: NotificationPageResponse) : this(
-            notificationPage.time,
-            notificationPage.hasMore,
-            notificationPage.notifications
-        )
-    }
-
-    data class CompleteList(override val time: String, override val hasMore: Boolean, override val notifications: List<EventResponse>) :
-        NotificationResponse() {
-        internal constructor(notificationPage: NotificationPageResponse) : this(
-            notificationPage.time,
-            notificationPage.hasMore,
-            notificationPage.notifications
-        )
-    }
-}
-
-@Serializable
-internal data class NotificationPageResponse(
+data class NotificationResponse(
     @SerialName("time") val time: String,
     @SerialName("has_more") val hasMore: Boolean,
-    @SerialName("notifications") val notifications: List<EventResponse>
+    @SerialName("notifications") val notifications: List<EventResponse>,
+    @Transient val isMissingNotifications: Boolean = false
```

Comment:
`isMissingNotifications` is not present in the response body, so `@Transient` is added here.

We flip the value to `true` when the server returns a `404`.